rs/utils/thread/src/deallocator_thread.rs

```diff
@@ -0,0 +1,101 @@
+use crate::JoinOnDrop;
```

Comment:
We should take this opportunity and fix the code ownership. How about assigning all of `rs/utils/thread` to the MR team?